### PR TITLE
Addition of Pattern and Gradient Types

### PR DESCRIPTION
**Pull request by ialdencoots**

### CODE REVIEW

File: src/Canvas.elm [hunk 1]
```diff
@@ -112,6 +115,33 @@ type DrawOp
...
+  = Repeat
+  | RepeatX
+  | RepeatY
+  | NoRepeat
+
+{-| A `Gradient` may be used to fill rectangles, lines, etc. The `LinearGradient` takes a starting `Point` for the gradient, an ending `Point` for the gradient, and the `ColorStop`s to apply. The `RadialGradient` takes the `Point` of the center of the starting circle, a `Float` representing its radius, the `Point` representing the center of the ending circle, a `Float` representing its radius, and the `ColorStop`s to apply.
+-}
+type Gradient
+    = LinearGradient Point Point (List ColorStop)
+    | RadialGradient Point Float Point Float (List ColorStop)
```
Comment (Chadtech) [inline src/Canvas.elm]: What do you think about doing it like this?
```elm

type Style
    = Color Color
    | Pattern Canvas Repeat
    | LinearGradient Point Point (List ColorStop)
    | RadialGradient Point Float Point Float (List ColorStop)
```

**Review by Chadtech — COMMENTED** [review]
Awesome! Thanks again for submitting this.

Moving forward tho, I think I would like to keep examples 0 and 1 as they were. They are the first ones, so they should be as basic and simple as possible. Do you mind reverting that?  An entirely new example around gradients might be cool.

File: src/Canvas.elm (hunk 1 above)
Comment (ialdencoots) [reply src/Canvas.elm]: Yeah, I guess that's a bit cleaner, isn't it? I'll make the change and revert the examples.

**Comment by ialdencoots** [comment]
I made the changes you suggested, reverted the examples and made a new one for the gradients.

**Comment by ialdencoots** [comment]
Oh man, I just discovered that the `Color` module has its own gradient type. Maybe we should update this to use that? http://package.elm-lang.org/packages/elm-lang/core/latest/Color#Gradient

**Comment by Chadtech** [comment]
Oh. Nice find! Yeah switching over to `Color.Gradient` seems like a good move to me.

**Comment by ialdencoots** [comment]
So I removed the `ColorStop` type and the `LinearGradient` and `RadialGradient` `Style` constructors, replacing them with a `Gradient Color.Gradient` constructor.

**Comment by Chadtech** [comment]
Great, I just tested everything and reviewed the code. Everything is looking good. Thanks again for contributing!

**Pull request merged**